test/integration/component/test_rootvolume_resize.py

```diff
@@ -0,0 +1,1140 @@
+                               list_storage_pools)
+from marvin.lib.utils import (cleanup_resources,
+                              validateList)
+from marvin.codes import (PASS,
+                          FAIL,
+                          FAILED,
+                          RESOURCE_PRIMARY_STORAGE,
+                          INVALID_INPUT)
+from marvin.lib.utils import checkVolumeSize
+import time
+from marvin.sshClient import SshClient
+
+
+class TestResizeVolume(cloudstackTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.testClient = super(TestResizeVolume, cls).getClsTestClient()
+        cls.api_client = cls.testClient.getApiClient()
+        cls.hypervisor = (cls.testClient.getHypervisorInfo()).lower()
+        cls.storageID = None
+        # Fill services from the external config file
+        cls.services = cls.testClient.getParsedTestDataConfig()
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.api_client)
+        cls.zone = get_zone(
+            cls.api_client,
+            cls.testClient.getZoneForTests())
+        cls.services["mode"] = cls.zone.networktype
+        cls._cleanup = []
+        cls.unsupportedStorageType = False
+        cls.unsupportedHypervisorType = False
+        cls.updateclone = False
+        if cls.hypervisor not in ['xenserver',"kvm","vmware"]:
+            cls.unsupportedHypervisorType=True
+            return
+        cls.template = get_template(
+            cls.api_client,
+            cls.zone.id
+        )
+        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
+        cls.services["virtual_machine"]["template"] = cls.template.id
+        cls.services["volume"]["zoneid"] = cls.zone.id
+        try:
+            cls.parent_domain = Domain.create(cls.api_client,
+                                              services=cls.services[
+                                                  "domain"],
+                                              parentdomainid=cls.domain.id)
+            cls.parentd_admin = Account.create(cls.api_client,
+                                               cls.services["account"],
+                                               admin=True,
+                                               domainid=cls.parent_domain.id)
+            cls._cleanup.append(cls.parentd_admin)
+            cls._cleanup.append(cls.parent_domain)
+            list_pool_resp = list_storage_pools(cls.api_client,
+                                               account=cls.parentd_admin.name,domainid=cls.parent_domain.id)
+            res = validateList(list_pool_resp)
+            if res[2]== INVALID_INPUT:
+                raise Exception("Failed to  list storage pool-no storagepools found ")
+            #Identify the storage pool type  and set vmware fullclone to true if storage is VMFS
+            if cls.hypervisor == 'vmware':
+                for strpool in list_pool_resp:
+                    if strpool.type == "VMFS":
```

Comment:
The same is the case for NFS type of primary storage. Is there a reason you are excluding NFS?